### DISCUSSION TITLE
Why VQVAE? How about VAE?

### DISCUSSION THREAD

**Discussion by LinB203** [PKU-YuanGroup/Open-Sora-Plan#18] [Q&A]
In fact, we believe that both VQVAE and VAE can be used, but we have noticed that there is already a Video-VQVAE in the community that can be directly used for testing purposes. 

Additionally, the combination of VQVAE and Diffusion works well for [Latent-Diffusion](https://github.com/CompVis/latent-diffusion?tab=readme-ov-file#pretrained-ldms). We have attempted to train a Video-VAE, but it was too resource-intensive. In the early stages of the Open-Sora Plan, the open-source Video-VQVAE can serve as a viable alternative.

**Comment by Edwardyan1112** [comment]
According to your estimates, how much computational resources and video datasets are required to train a Video VAE (not Video-VQVAE)? Approximately, how long would the training process take?

**Reply by LinB203** [reply]
We have tested that it can only batch size 2 on 80G. Input 16 frames x 256 x 256 and latent size 2 x 32 x 32.
We are not sure when it will converge because it is just too costly.

**Comment by lucasjinreal** [comment]
Does Sora used VAE not VQVAE?

**Reply by LinB203** [reply]
It's unknown, and at the moment we use Video-VQVAE as just a compromise.

**Reply by lucasjinreal** [reply]
Maybe VQVAE can handle it, stabledifusion3 also using VQVAE

**Reply by CuddleSabe** [reply]
Maybe just another ViT as encoder

**Comment by ttttt12345** [comment]
it is unreal bro,dont waste your time on this project.

**Comment by LinB203** [comment]
We are now able to support both vae and vqvae, please check our latest code.

**Reply by lucasjinreal** [reply]
how's the performance compare

**Comment by LinB203** [comment]
Refer to https://github.com/PKU-YuanGroup/Open-Sora-Plan/issues/93 for more details.

**Comment by hamza-m-farooqi** [comment]
looking at the intense resource requirements for Video-VAE and the practicality of Video-VQVAE, it seems like a smart compromise for now. eager to see how this compromise  plays out 😊
